client.go

```diff
@@ -82,11 +83,18 @@ func (c *Client) do(method, path string, body interface{}, result interface{}) e
 	if !isHTTPSuccess(resp.StatusCode) {
 		defer resp.Body.Close()
 		body, _ := ioutil.ReadAll(resp.Body)
+
+		// Dump request headers sans Authorization.
+		var b bytes.Buffer
+		writer := bufio.NewWriter(&b)
```

Comment:
NBD but I believe you can use the newer `strings.Builder` (and then define `reqHeaders` as `string`) to save some allocs. It implements `io.Writer`, and also implements the internal `writeStringer` interface that [`WriteSubset` uses](https://golang.org/src/net/http/header.go#L148).

I'm not sure, but you might be able to do the same thing for the body too. Something like:
```
var body strings.Builder
_ = io.Copy(body, resp.Body)
body.String()
```

Honestly this doesn't matter, since this code path is only called on rare errors. So feel free to disregard. I just had fun reading the stdlib documentation 😍

Reply:
i like it